go/vt/vttablet/tabletmanager/vreplication/controller.go

```diff
@@ -178,9 +185,18 @@ func (ct *controller) runBlp(ctx context.Context) (err error) {
 
 		player := binlogplayer.NewBinlogPlayerTables(dbClient, tablet, tables, ct.id, ct.blpStats)
 		return player.ApplyBinlogEvents(ctx)
+	case ct.source.KeyRange != nil:
+		player := binlogplayer.NewBinlogPlayerKeyRange(dbClient, tablet, ct.source.KeyRange, ct.id, ct.blpStats)
+		return player.ApplyBinlogEvents(ctx)
+	case ct.source.Filter != nil:
```

Comment:
I think we are missing updating `legacy_split_clone` and `split_clone` vreplication exec command so it provides a filter instead of a KeyRange. That we way we can test shard splits with the new player.

Reply:
Ah ok. That makes sense.